lara-typescript/src/interactive-api-client/index.ts

```diff
@@ -0,0 +1,344 @@
+    features: {
+      aspectRatio?: string;
+      authoredState?: boolean;
+    };
+  }
+
+  export interface INavigationOptions {
+    enableForwardNav?: boolean;
+    message?: string;
+  }
+
+  export interface IGetFirebaseJwtOptions {
+    firebase_app?: string;
+  }
+
+  export interface IFirebaseJwtResponse {
+    response_type?: "ERROR";
+    message?: string;
+    token?: string;
+  }
+
+  export interface IClientOptions<AuthoredState = {}, InteractiveState = {}, GlobalInteractiveState = {}> {
+    name: string;
+    verbosity?: VerbosityLevel;
+    startDisconnected?: boolean;
+    supportedFeatures?: ISupportedFeatures;
+    onHello?: () => void;
+    onLoadInteractive?: (interactiveState: InteractiveState | null) => void;
+    // tslint:disable-next-line:max-line-length
+    onInitInteractive?: (initMessage: IInitInteractive<AuthoredState, InteractiveState, GlobalInteractiveState>) => void;
+    onGetLearnerUrl?: () => string | null;
+    onGetInteractiveState?: () => InteractiveState | string | null;
+  }
+
+  export const InIframe = inIframe;
+
+  export class Client<AuthoredState = {}, InteractiveState = {}, GlobalInteractiveState = {}> {
+
+    public static GetLocalVerbosityLevel(name: string) {
+      const level =
+        window.localStorage.getItem(`LaraInteractiveApiVerbosity[${name}]`) ||
+        window.localStorage.getItem(`LaraInteractiveApiVerbosity`);
+      return level !== null ? level as VerbosityLevel : undefined;
+    }
+
+    private phone: iframePhone.IFrameEndpoint | undefined;
+    private options: IClientOptions<AuthoredState, InteractiveState, GlobalInteractiveState>;
+
+    constructor(options: IClientOptions<AuthoredState, InteractiveState, GlobalInteractiveState>) {
+      options.verbosity = Client.GetLocalVerbosityLevel(options.name) || options.verbosity;
+      this.options = options;
+      this.log("debug", "constructor: verbosity level is", options.verbosity);
+
+      if (inIframe) {
+        this.log("debug", "constructor: in iframe");
+        if (!options.startDisconnected) {
+          this.log("debug", "constructor: automatically connecting");
+          this.connect();
+        }
+      } else {
+        this.log("debug", "constructor: not in iframe");
+      }
+    }
+
+    public connect() {
+      if (inIframe) {
+        if (!this.phone) {
+          this.log("debug", "#connect: connecting");
+          this.phone = iframePhone.getIFrameEndpoint();
+
+          this.addListener("hello", () => {
+            if (this.options.onHello) {
+              this.log("debug", "onHello listener called, calling client callback");
+              this.options.onHello();
+            } else {
+              this.log("debug", "onHello listener called, no client callback found");
+            }
+
+            if (this.options.supportedFeatures) {
+              this.setSupportedFeatures(this.options.supportedFeatures);
+            }
+          });
+
+          this.addListener("getLearnerUrl", () => {
```

Comment:
This is deprecated and strongly discouraged. I think it should be removed from the client. If an interactive is upgraded to use the client, then it should also be upgraded to no use deprecated messages like this.